rust/operator/src/container.rs

```diff
@@ -391,26 +482,31 @@ impl ContainerConfig {
                 // for e.g. scaling. It may fail if the active namenode is restarted and the standby
                 // namenode takes over.
                 // This is why in the second part we check if the node is formatted already via
-                // $NAMENODE_DIR/current/VERSION. Then we dont do anything.
+                // $NAMENODE_DIR/current/VERSION. Then we don't do anything.
                 // If there is no active namenode, the current pod is not formatted we format as
                 // active namenode. Otherwise as standby node.
-                args.push(formatdoc!(r###"
+                if hdfs.has_kerberos_enabled() {
+                    args.push(Self::get_kerberos_ticket(hdfs, role, object_name)?);
+                }
+                args.push(formatdoc!(
+                    r###"
+                    cat "{NAMENODE_ROOT_DATA_DIR}/current/VERSION"
                     echo "Start formatting namenode $POD_NAME. Checking for active namenodes:"
-                    for id in {pod_names}
+                    for namenode_id in {pod_names}
                     do
-                      echo -n "Checking pod $id... "
-                      SERVICE_STATE=$({hadoop_home}/bin/hdfs haadmin -getServiceState $id 2>/dev/null)
+                      echo -n "Checking pod $namenode_id... "
+                        {get_service_state_command}
                       if [ "$SERVICE_STATE" == "active" ]
                       then
-                        ACTIVE_NAMENODE=$id
+                        ACTIVE_NAMENODE=$namenode_id
                         echo "active"
                         break
                       fi
                       echo ""
                     done
 
                     set -e
-                    if [ ! -f "{NAMENODE_ROOT_DATA_DIR}/current/VERSION" ]
+                    if ! ls {NAMENODE_ROOT_DATA_DIR}/current/fsimage_*
```

Comment:
Maybe this causes the JN problems. @nightkr *only if you have the time and inclination*: This from your spike, what was the intention of changing this?

Reply:
Ok, didn't change anything :/

Reply:
Would probably cause breakage when upgrading the operator later on though, similar to when we renamed datanode paths.

Reply:
Would it? This only changes the command in the init container